### PR TITLE
Rewrite Inbound SMS Guide

### PR DESCRIPTION
**Pull request by marklewin**
## Description

General tidy up and rewrite.

## Deploy Notes

Check in sample app. No dependencies on building blocks, etc.

### CODE REVIEW

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 1]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
+`type` | Possible values are: <ul><li>`text` - standard text</li><li>`unicode` - a text message that contains Unicode characters and is encoded using the `application/x-www-form-urlencoded` type - see [definition](https://en.wikipedia.org/wiki/Percent-encoding#The_application.2Fx-www-form-urlencoded_type)<li>`binary` - a binary message</li></ul>| Yes
+`to` | The phone number that the message was sent *to*. **This is your virtual number**.| Yes
+`msisdn` | The phone number that this inbound message was sent *from*. | Yes
+`messageId` | Nexmo's unique identifier for this message.| Yes
+`message-timestamp` | The [UTC±00:00](https://en.wikipedia.org/wiki/UTC%C2%B100:00) time when Nexmo started to push this inbound message to your webhook endpoint, in the following format: `YYYY-MM-DD HH:MM:SS`.| Yes
+`timestamp` | The [unix timestamp](https://www.unixtimestamp.com/) representation of `message-timestamp`. | If your messages are [signed](/concepts/guides/signing-messages)
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: Message signing doesn't affect timestamp. Looks like a copy/paste error?

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 2]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
-`message-timestamp` | The time at [UTC±00:00](https://en.wikipedia.org/wiki/UTC%C2%B100:00) that Nexmo started to push this inbound message to your [webhook endpoint](/concepts/guides/webhooks). The *message-timestamp* is in the following format *YYYY-MM-DD HH:MM:SS*. For example, *2020-01-01 12:00:00*.| Yes
-`timestamp` | A unix timestamp representation of *message-timestamp*. | If your messages are [signed](/concepts/guides/signing-messages)
-`nonce` | A random string that forms part of the signed set of parameters, it adds an extra element of unpredictability into the signature for the request. You use the *nonce* and *timestamp* parameters with your shared secret to calculate and validate the signature for inbound messages. | If your messages are [signed](/concepts/guides/signing-messages)
+`type` | Possible values are: <ul><li>`text` - standard text</li><li>`unicode` - a text message that contains Unicode characters and is encoded using the `application/x-www-form-urlencoded` type - see [definition](https://en.wikipedia.org/wiki/Percent-encoding#The_application.2Fx-www-form-urlencoded_type)<li>`binary` - a binary message</li></ul>| Yes
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: This line is confusing to me. Why is unicode encoded and text isn't? They're the same format, just different encodings

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 3]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
+`to` | The phone number that the message was sent *to*. **This is your virtual number**.| Yes
+`msisdn` | The phone number that this inbound message was sent *from*. | Yes
+`messageId` | Nexmo's unique identifier for this message.| Yes
+`message-timestamp` | The [UTC±00:00](https://en.wikipedia.org/wiki/UTC%C2%B100:00) time when Nexmo started to push this inbound message to your webhook endpoint, in the following format: `YYYY-MM-DD HH:MM:SS`.| Yes
+`timestamp` | The [unix timestamp](https://www.unixtimestamp.com/) representation of `message-timestamp`. | If your messages are [signed](/concepts/guides/signing-messages)
+`nonce` | A random string that adds an extra element of unpredictability into the signature for the request. You use the `nonce` and `timestamp` parameters with your shared secret to calculate and validate the signature for inbound messages. | If your messages are [signed](/concepts/guides/signing-messages)
 
 
-**For an inbound message**
+### For messages of type `text`
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: Should we show `unicode` examples too?

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 4]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
-`text` | The message body for this inbound message.| If *type* is `text` |
-`keyword` | The first word in the message body. This is typically used with short codes. | If *type* is `text` |
+`text` | The message body for this inbound message.| If `type` is `text` |
+`keyword` | The first word in the message body. This is typically used with short codes. | If `type` is `text` |
 
 
-**For *concatenated* inbound messages**
+### For concatenated inbound messages
+
+If a message sent to your virtual number is longer than the maximum number of characters allowed in a single message, `concat` is `true` and you receive the message in parts.
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: I'd highlight that if it's a single part message, `concat` will not exist

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 5]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
-`concat-total` | The number of parts in this concatenated message.| If *concat* is *True*
-`concat-part` | The number of this part in the message. Counting starts at *1*. | If *concat* is *True*
+`concat` | `true` if this is a concatenated message. | Yes
+`concat-ref` | The transaction reference. All parts of this message share this `concat-ref`.| If `concat` is `true`
+`concat-total` | The number of parts in this concatenated message.| If `concat` is `true`
+`concat-part` | The number of this part in the message. The first part of the message is `1`. | If `concat` is `true`
 
 
-**For binary messages**
+### For messages of type `binary`
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: This should be above concatenated messages imo

File: _documentation/messaging/sms/guides/inbound-sms.md [hunk 6]
```diff
@@ -4,47 +4,52 @@ title: Inbound SMS
...
 
 
-**For an inbound message**
+### For messages of type `text`
 
 Key | Value | Required
 -- | -- | --
-`text` | The message body for this inbound message.| If *type* is `text` |
-`keyword` | The first word in the message body. This is typically used with short codes. | If *type* is `text` |
+`text` | The message body for this inbound message.| If `type` is `text` |
```
Comment (mheap) [inline _documentation/messaging/sms/guides/inbound-sms.md]: I'm not sure how much value the `Required` field adds, as we're already under a `text` type header

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 1 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Good spot. Yes it was!

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 2 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Agreed. Removed and just listed the different permitted values.

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 4 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Added.

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 5 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Agreed and moved.

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 6 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Yes, agreed. Removed.

File: _documentation/messaging/sms/guides/inbound-sms.md (hunk 3 above)
Comment (marklewin) [reply _documentation/messaging/sms/guides/inbound-sms.md]: Yes. The request body is the same, but made sure to mention it.

**Pull request merged**